app/main.py

```diff
@@ -1 +1,32 @@
-# write your code here
+class Animal:
+    alive = []  # Class attribute to track all alive animals
+
+    def __init__(self, name, health=100):
+        self.name = name
+        self.health = health
+        self.hidden = False
+        if self.health > 0:
+            Animal.alive.append(self)
+
+    def __repr__(self):
+        return f"{{Name: {self.name}, Health: {self.health}, Hidden: {self.hidden}}}"
+
+    @classmethod
+    def update_alive(cls):
+        cls.alive = [animal for animal in cls.alive if animal.health > 0]
+
+
+class Herbivore(Animal):
+    def hide(self):
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+    def bite(self, prey):
+        if isinstance(prey, Herbivore) and not prey.hidden:
```

Comment:
The condition `isinstance(prey, Herbivore)` assumes that only `Herbivore` instances can be prey. If there are other subclasses of `Animal` that can be prey, this condition should be adjusted accordingly.